csv/src/fs2/data/csv/package.scala

```diff
@@ -48,18 +52,64 @@ package object csv {
   def skipHeaders[F[_]]: Pipe[F, NonEmptyList[String], Row] =
     _.tail.map(new Row(_))
 
-  def decode[F[_], R](implicit F: RaiseThrowable[F], R: RowDecoder[R]): Pipe[F, Row, R] =
+  def decode[F[_], R](implicit F: RaiseThrowable[F],
+                      R: RowDecoder[R]): Pipe[F, Row, R] =
     _.map(row => R(row.values)).rethrow
 
-  def attemptDecode[F[_], R](implicit R: RowDecoder[R]): Pipe[F, Row, DecoderResult[R]] =
+  def attemptDecode[F[_], R](
+      implicit R: RowDecoder[R]): Pipe[F, Row, DecoderResult[R]] =
     _.map(row => R(row.values))
 
-  def decodeRow[F[_], Header, R](implicit F: RaiseThrowable[F],
-                                 R: CsvRowDecoder[R, Header]): Pipe[F, CsvRow[Header], R] =
+  def decodeRow[F[_], Header, R](
+      implicit F: RaiseThrowable[F],
+      R: CsvRowDecoder[R, Header]): Pipe[F, CsvRow[Header], R] =
     _.map(R(_)).rethrow
 
-  def attemptDecodeRow[F[_], Header, R](
-      implicit R: CsvRowDecoder[R, Header]): Pipe[F, CsvRow[Header], DecoderResult[R]] =
+  def attemptDecodeRow[F[_], Header, R](implicit R: CsvRowDecoder[R, Header])
+    : Pipe[F, CsvRow[Header], DecoderResult[R]] =
     _.map(R(_))
 
+  def writeWithHeaders[F[_], Header](headers: NonEmptyList[Header])(
+      implicit H: WriteableHeader[Header]): Pipe[F, Row, NonEmptyList[String]] =
+    Stream(H(headers)) ++ _.map(_.values)
+
+  def writeWithoutHeaders[F[_]]: Pipe[F, Row, NonEmptyList[String]] =
+    _.map(_.values)
+
+  def toStrings[F[_]](
+      separator: Char = ',',
+      newline: String = "\n"): Pipe[F, NonEmptyList[String], String] = {
+    _.flatMap(
+      nel =>
+        Stream
+          .emits(nel.toList)
+          .map(RowWriter.encodeColumn(separator))
+          .intersperse(separator.toString) ++ Stream(newline))
+  }
+
+  def toRowStrings[F[_]](
+      separator: Char = ',',
+      newline: String = "\n"): Pipe[F, NonEmptyList[String], String] = {
+    // explicit Show avoids mapping the NEL before
+    val showColumn: Show[String] =
+      Show.show(RowWriter.encodeColumn(separator))
```

Comment:
I would make escaping optional, you might have a stream that is already escaped, or you know you don't need it.

Reply:
Good point. I took this a little further and introduced an `EscapeMode` enum which allows to rely on automatic escaping, force escaping or completely turn it off.